tasks/main.yml

```diff
@@ -1,103 +1,123 @@
+    - mongo
 
 - name: remove apt repository 3.2
   apt_repository:
-    repo: "deb http://repo.mongodb.org/apt/ubuntu {{ ansible_distribution_release}}/mongodb-org/3.2 multiverse"
+    repo: 'deb http://repo.mongodb.org/apt/ubuntu {{ ansible_distribution_release}}/mongodb-org/3.2 multiverse'
     state: absent
   when: mongodb_version != "3.2" and ansible_distribution == 'Ubuntu'
   tags:
-  - mongo
+    - mongo
 
 - name: add apt repository 3.4
   apt_repository:
-    repo: "deb [ arch=amd64 ] http://repo.mongodb.org/apt/ubuntu {{ ansible_distribution_release}}/mongodb-org/3.4 multiverse"
+    repo: 'deb [ arch=amd64 ] http://repo.mongodb.org/apt/ubuntu {{ ansible_distribution_release}}/mongodb-org/3.4 multiverse'
     state: present
   when: mongodb_version == "3.4" and ansible_distribution == 'Ubuntu'
   register: mongodb_repo_added
   tags:
-  - mongo
+    - mongo
 
 - name: remove apt repository 3.4
   apt_repository:
-    repo: "deb [ arch=amd64 ] http://repo.mongodb.org/apt/ubuntu {{ ansible_distribution_release}}/mongodb-org/3.4 multiverse"
+    repo: 'deb [ arch=amd64 ] http://repo.mongodb.org/apt/ubuntu {{ ansible_distribution_release}}/mongodb-org/3.4 multiverse'
     state: absent
   when: mongodb_version != "3.4" and ansible_distribution == 'Ubuntu'
   tags:
-  - mongo
+    - mongo
 
 - name: add apt repository 3.6
   apt_repository:
-    repo: "deb [ arch=amd64 ] http://repo.mongodb.org/apt/ubuntu {{ ansible_distribution_release}}/mongodb-org/3.6 multiverse"
+    repo: 'deb [ arch=amd64 ] http://repo.mongodb.org/apt/ubuntu {{ ansible_distribution_release}}/mongodb-org/3.6 multiverse'
     state: present
   when: mongodb_version == "3.6" and ansible_distribution == 'Ubuntu'
   register: mongodb_repo_added
   tags:
-  - mongo
+    - mongo
 
 - name: remove apt repository 3.6
   apt_repository:
-    repo: "deb [ arch=amd64 ] http://repo.mongodb.org/apt/ubuntu {{ ansible_distribution_release}}/mongodb-org/3.6 multiverse"
+    repo: 'deb [ arch=amd64 ] http://repo.mongodb.org/apt/ubuntu {{ ansible_distribution_release}}/mongodb-org/3.6 multiverse'
     state: absent
   when: mongodb_version != "3.6" and ansible_distribution == 'Ubuntu'
   tags:
-  - mongo
+    - mongo
+
+- name: add apt repository 4.2
+  apt_repository:
+    repo: 'deb [ arch=amd64 ] http://repo.mongodb.org/apt/ubuntu {{ ansible_distribution_release}}/mongodb-org/4.2 multiverse'
+    state: present
+  when: mongodb_version == "4.2" and ansible_distribution == 'Ubuntu'
+  register: mongodb_repo_added
+  tags:
+    - mongo
+
+- name: remove apt repository 4.2
+  apt_repository:
+    repo: 'deb [ arch=amd64 ] http://repo.mongodb.org/apt/ubuntu {{ ansible_distribution_release}}/mongodb-org/4.2 multiverse'
+    state: absent
+  when: mongodb_version != "4.2" and ansible_distribution == 'Ubuntu'
+  tags:
+    - mongo
 
 - name: apt update cache
   apt:
     update_cache: yes
   when: mongodb_repo_added.changed
   tags:
-  - mongo
+    - mongo
+
+- name: uninstall mongodb
```

Comment:
This step may be really dangerous

Reply:
How so? Do you think apt would remove data when removing mongodb ?